spec/requests/api/snapshots_spec.rb

```diff
@@ -262,6 +279,29 @@
         expect(response.parsed_body).to include(expected)
         expect(response).to have_http_status(:ok)
       end
+
+      it "raises a 404 with proper message if a resource isn't found" do
+        api_basic_authorize(action_identifier(:vms, :delete, :snapshots_subresource_actions, :delete))
```

Comment:
I think you want the `:post` action here, and `:snaphots_subcollection_actions` (looks like this might be wrong in some of the examples above)